libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -574,7 +604,8 @@ AP_GPS::update_instance(uint8_t instance)
         return;
     }
 
-    if (drivers[instance] == nullptr || state[instance].status == NO_GPS) {
+    // If it is UAVCAN GNSS - give it regular time as if it is there to acknowledge itself
+    if (drivers[instance] == nullptr || (state[instance].status == NO_GPS && _type[instance] != GPS_TYPE_UAVCAN)) {
```

Comment:
Old PX4 UAVCAN GPS suffered from same, it got reinitialized very often until fix was found. I will change as you requested, so this behavior will remain now for new drivers.